### ISSUE TITLE
TT-162 Error in virtual endpoint when using JSVM with key created from portal

### ISSUE THREAD

**Issue by n2linux** [TykTechnologies/tyk#3218]
**Branch/Environment**
- Branch: 2.9, Master
- Environment: On-prem

**Describe the bug**
Error in virtual endpoint when using JSVM with key created from portal

**Reproduction steps**
1. Create virtual endpoint
2. Use the function below as the JS function to call (taken from our docs)
3. Set 'use_session:true' in the API definition
4. Create a key to access the API
5. Everything works
6. Create a key through the developer portal and use it to access the API
7. Receive error message

**Actual behavior**
When accessing the API with a developer token, I receive the following error message:
```
[Jul 17 12:16:59] ERROR Failed to decode virtual endpoint response data on return from VM: ; Returned: {"Response":{"Body":"yo yo","Code":200,"Headers":{"location":"[\"/get\"]","test":"virtual","test-2":"virtual"}},"SessionMeta":{"tyk_developer_id":"5f11cc1ba4b16a176b4a6735","tyk_key_request_fields":{},"tyk_user_fields":{}}} api_id=a5d336a305984587482dfcb2cb95133c api_name=Auth token error=json: cannot unmarshal object into Go struct field VMResponseObject.SessionMeta of type string
```

Both the output of the `log("Session Metadata:"+ JSON.stringify(session.meta_data))` in the function and the error message indicate that there is an extra `}` in the session.meta_data:
```
"SessionMeta":{"tyk_developer_id":"5f11cc1ba4b16a176b4a6735","tyk_key_request_fields":{},"tyk_user_fields":{}}}
```

**Expected behavior**
I expect the call to return the proper response including the session metadata

**Additional context**
Raised by a customer 

JS function:
```javascript
function myVirtualHandlerGetHeaders (request, session, config) {
  rawlog("Virtual Test running")
    
  //Usage examples:
  log("Request Session: " + JSON.stringify(session))
  log("API Config:" + JSON.stringify(config))
  log("Session Metadata:"+ JSON.stringify(session.meta_data))
  
  log("Request object: " + JSON.stringify(request))   
  log("Request Body: " + JSON.stringify(request.Body))
  log("Request Headers:" + JSON.stringify(request.Headers))
  log("param-1:" + request.Params["param1"])
    
  log("Request header type:" + typeof JSON.stringify(request.Headers))
  log("Request header:" + JSON.stringify(request.Headers.Location))
    

  //Make api call to upstream target
  newRequest = {
    "Method": "GET",
    "Body": "",
    "Headers": {"location":JSON.stringify(request.Headers.Location)},
    "Domain": "http://httpbin.org",
    "Resource": "/headers",
    "FormData": {}
  };
  rawlog("--- before get to upstream ---")
  response = TykMakeHttpRequest(JSON.stringify(newRequest));
  rawlog("--- After get to upstream ---")
  log("response type: " + typeof response);
  log("response: " + response);
  usableResponse = JSON.parse(response);
  var bodyObject = JSON.parse(usableResponse.Body);
    
  var responseObject = {
    //Body: "THIS IS A  VIRTUAL RESPONSE",
    Body: "yo yo",
    Headers: {
      "test": "virtual",
      "test-2": "virtual",
      "location" : bodyObject.headers.Location
    },
    Code: usableResponse.Code
  }
    
  rawlog("Virtual Test ended")
  return TykJsResponse(responseObject, session.meta_data)   
}
```

**Comment by sedkis** [comment]
Confirmed with an even simpler JS Post hook:
```
var exampleJavaScriptMiddlewarePostHook = new TykJS.TykMiddleware.NewMiddleware({});

exampleJavaScriptMiddlewarePostHook.NewProcessRequest(function(request, session, spec) {

  log('----------------------')
  log('session meta: ')
  log(JSON.stringify(session.meta_data))
   
  return exampleJavaScriptMiddlewarePostHook.ReturnData(request, session.meta_data);
});
```

**Comment by agata-wit** [comment]
Fixed and released in Tyk 3.1.1 and Tyk 3.0.2
